model/src/application/interaction/application_command/data/mod.rs

```diff
@@ -125,24 +81,250 @@ impl Serialize for CommandDataOption {
+                                    "expected integer got {:?}",
+                                    other
+                                )));
+                            }
+                        }
+                    }
+                    CommandOptionType::Integer => {
+                        let val = value_opt.ok_or_else(|| DeError::missing_field("value"))?;
+
+                        if let ValueEnvelope::Integer(i) = val {
+                            CommandOptionValue::Integer(i)
+                        } else {
+                            return Err(DeError::custom(&format!(
+                                "expected integer got {:?}",
+                                val
+                            )));
+                        }
+                    }
+                    CommandOptionType::Mentionable => {
+                        let val = value_opt.ok_or_else(|| DeError::missing_field("value"))?;
+
+                        match val {
+                            ValueEnvelope::Integer(i) => {
+                                let id = GenericId::new(i.try_into().map_err(|_| {
+                                    DeError::custom("could not convert signed integer to unsigned")
+                                })?)
+                                .ok_or_else(|| DeError::custom("mentionable id was zero"))?;
+
+                                CommandOptionValue::Mentionable(id)
+                            }
+                            ValueEnvelope::Id(id) => CommandOptionValue::Mentionable(id),
+                            other => {
+                                return Err(DeError::custom(&format!(
+                                    "expected integer got {:?}",
+                                    other
+                                )));
+                            }
+                        }
+                    }
+                    CommandOptionType::Number => {
+                        let val = value_opt.ok_or_else(|| DeError::missing_field("value"))?;
+
+                        match val {
+                            ValueEnvelope::Integer(i) => {
+                                // As json allows sending floating
+                                // points without the tailing decimals
+                                // it may be interpreted as a integer
+                                // but it is safe to cast as there can
+                                // not occur any loss.
+                                #[allow(clippy::cast_precision_loss)]
+                                CommandOptionValue::Number(Number(i as f64))
+                            }
+                            ValueEnvelope::Number(f) => CommandOptionValue::Number(Number(f)),
+                            _ => {
+                                return Err(DeError::custom(&format!(
+                                    "expected double got {:?}",
+                                    val
+                                )));
```

Comment:
Is this intentional?
```suggestion
                            other => {
                                return Err(DeError::custom(&format!(
                                    "expected double got {:?}",
                                    other
                                )));
```

Reply:
I don't think this was resolved